qiskit_nature/second_q/circuit/library/ansatzes/ucc.py

```diff
@@ -293,17 +292,22 @@ def operators(self):  # pylint: disable=invalid-overridden-method
                 # the ``excitation_list`` is transformed identically to the operators, we retain
                 # ``None`` for non-commuting operators in order to manually remove them in unison.
                 operators = self.qubit_converter.convert_match(excitation_ops, suppress_none=False)
-                valid_operators, valid_excitations = [], []
-                for op, ex in zip(operators, self._excitation_list):
-                    if op is not None:
-                        valid_operators.append(op)
-                        valid_excitations.append(ex)
-
-                self._excitation_list = valid_excitations
-                self.operators = valid_operators
+                self._check_length_excitations_list(
+                    operators=operators, excitation_list=self._excitation_list
+                )
 
         return super(UCC, self.__class__).operators.__get__(self)
 
+    def _check_length_excitations_list(self, operators, excitation_list):
```

Comment:
```suggestion
    def _filter_operators(self, operators, excitation_list):
```
This method is really filtering the operators as explained in the comment a few lines above the original code:
> The converted operators are maintained at the same index by the converter
> inserting ``None`` as the result if an operator did not commute. To ensure that
> the ``excitation_list`` is transformed identically to the operators, we retain
> ``None`` for non-commuting operators in order to manually remove them in unison.

This naming reflects better, that this method actually changes objects on `self`, which would have been surprising to me in the old name.